pkg/autodiscovery/providers/config_reader.go

```diff
@@ -325,77 +327,87 @@ func collectDir(parentPath string, folder os.FileInfo, integrationErrors map[str
 // GetIntegrationConfigFromFile returns an instance of integration.Config if `fpath` points to a valid config file
 func GetIntegrationConfigFromFile(name, fpath string) (integration.Config, error) {
 	cf := configFormat{}
-	config := integration.Config{Name: name}
+	conf := integration.Config{Name: name}
 
 	// Read file contents
 	// FIXME: ReadFile reads the entire file, possible security implications
 	yamlFile, err := readFilePtr(fpath)
 	if err != nil {
-		return config, err
+		return conf, err
 	}
 
 	// Parse configuration
 	// Try UnmarshalStrict first, so we can warn about duplicated keys
 	if strictErr := yaml.UnmarshalStrict(yamlFile, &cf); strictErr != nil {
 		if err := yaml.Unmarshal(yamlFile, &cf); err != nil {
-			return config, err
+			return conf, err
 		}
 		log.Warnf("reading config file %v: %v\n", fpath, strictErr)
 	}
 
 	// If no valid instances were found & this is neither a metrics file, nor a logs file
 	// this is not a valid configuration file
 	if cf.MetricConfig == nil && cf.LogsConfig == nil && len(cf.Instances) < 1 {
-		return config, errors.New("Configuration file contains no valid instances")
+		return conf, errors.New("Configuration file contains no valid instances")
 	}
 
 	// at this point the Yaml was already parsed, no need to check the error
 	if cf.InitConfig != nil {
 		rawInitConfig, _ := yaml.Marshal(cf.InitConfig)
-		config.InitConfig = rawInitConfig
+		conf.InitConfig = rawInitConfig
 	}
 
 	// Go through instances and return corresponding []byte
 	for _, instance := range cf.Instances {
 		// at this point the Yaml was already parsed, no need to check the error
 		rawConf, _ := yaml.Marshal(instance)
-		config.Instances = append(config.Instances, rawConf)
+		dataConf := (integration.Data)(rawConf)
+		if fargate.IsFargateInstance(context.TODO()) {
```

Comment:
I would have preferred that, that autodiscovery doesn't have any specific logic around the environment detection. And delegate it to the agent initialisation or call the tagger to get "global" tags, but for now we don't have this entry in the tagger.

We can do it like this, and plan to restructure it in agent 7.36.

Reply:
Agreed, I'll create a card to restructure this for 7.36.

Reply:
(internal follow up card: AC-1576)